gke-deploy/core/container/container_test.go

```diff
@@ -15,35 +15,215 @@ package container
 import (
 	"context"
 	"fmt"
+	"reflect"
 	"testing"
 
+	"github.com/google/go-containerregistry/pkg/name"
+
 	"github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/testservices"
 )
 
-func TestGetDigest(t *testing.T) {
+func TestParseImages(t *testing.T) {
 	ctx := context.Background()
-	image := "my-image:1.0.0"
-	gs := &testservices.TestGcloud{
-		ContainerImagesDescribeResp: "sha256:foobar",
-		ContainerImagesDescribeErr:  nil,
+
+	image := "gcr.io/my-project/my-image:1.0.0"
+	image2 := "gcr.io/my-project/my-image-2@sha256:929665b8eb2bb286535d29cd73c71808d7e1ad830046333f6cf0ce497996eb79"
+	image3 := "gcr.io/my-project/my-image-3"
+
+	tests := []struct {
+		name string
+
+		images []string
+
+		want []Image
+	}{
```

Comment:
nit: To preserve vertical and horizontal space, you can put these `{`s on the same line, e.g.:

```go
tests := []struct{
  ...struct definition...
}{{
  ...case...
}, {
  ...case...
}, {
  ...case...
}}
```

Reply:
I agree that this will be better. If it's okay with you, I'd like to do this in a future PR in order to touch all existing tests and not pollute this PR.